internal/bsr/testing.go

```diff
@@ -3,13 +3,67 @@
 
 package bsr
 
+import (
+	"testing"
+	"time"
+
+	"io"
+
+	"github.com/stretchr/testify/require"
+)
+
 func TestSessionRecordingMeta(s string, p Protocol) *SessionRecordingMeta {
 	return &SessionRecordingMeta{
 		Id:       s,
 		Protocol: p,
 	}
 }
 
+func TestSessionRecordingSummary(s string) *BaseSessionSummary {
+	return &BaseSessionSummary{
+		Id:              s,
+		ConnectionCount: 1,
+		StartTime:       time.Now(),
+		EndTime:         time.Now(),
+	}
+}
+
+func TestConnectionRecordingSummary(s string) *BaseConnectionSummary {
+	return &BaseConnectionSummary{
+		Id:           s,
+		ChannelCount: 1,
+		StartTime:    time.Now(),
+		EndTime:      time.Now(),
+		BytesUp:      256,
+		BytesDown:    256,
+	}
+}
+
+func TestConnectionRecordingMeta(s string) *ConnectionRecordingMeta {
+	return &ConnectionRecordingMeta{
+		Id: s,
+		channels: map[string]bool{
+			"test": true,
+		},
+	}
+}
+
+func TestChannelRecordingMeta(s string) *ChannelRecordingMeta {
+	return &ChannelRecordingMeta{
+		Id:   s,
+		Type: "chan",
+	}
+}
```

Comment:
Similarly I am also not sure if these are helpful test helpers. In particular I am not sure if these are good defaults for external tests to use, and the helpers don't allow for easily providing alternatives, instead the code would need to set fields on the returned struct, which in some cases can not be set since the fields are not exported.